ow_lander/scripts/path_planning_commander.py

```diff
@@ -194,8 +205,29 @@ def run(self):
     self.guarded_move_srv = rospy.Service(
         'arm/guarded_move', GuardedMove, self.handle_guarded_move)
     print("path_planning_commander has started!")
+
     rospy.spin()
 
+  def stop_arm(self):
+    plan = activity_guarded_move.stop_guarded_move_plan(self.arm_move_group)
+    self.trajectory_async_executer.systemFaultListener(plan.joint_trajectory, self.handle_guarded_move_feedback)
+
+  def callback(self, data):
+    """
+    If system fault occurs, and it is an arm failure, path planning stops
+    """
+    if data.value == 4:
+      self.arm_fault = True
+      self.trajectory_async_executer.stop()
+    else:
+      self.arm_fault = False
+
+  def systemFaultListener(self):
+    """
+    Listens for system faults
+    """
+    rospy.Subscriber("/system_faults_status", SystemFaults, self.callback)
+
```

Comment:
To keep your code consistent with the rest of the file I recommend you move the subscription definition into the _run()_ method along with the rest of the subscribers. This would make it easier to review what services, publishers and subscribers that this class is making use of.

Reply:
To elaborate more, there is really no need to have this method that merely invokes a single line.

Reply:
fixed!